src/status_im/transport/handlers.cljs

```diff
@@ -185,3 +181,84 @@
+  {:db (update db :transport/message-envelopes dissoc envelope-hash)})
+
+(defn update-send-contact-request [chat-id {:keys [db] :as cofx}]
+  (let [chat         (get-in db [:transport/chats chat-id])
+        updated-chat (assoc chat :send-contact-request? false)]
+    {:db            (assoc-in db
+                              [:transport/chats chat-id :send-contact-request?]
+                              false)
+     :data-store/tx [(transport-store/save-transport-tx {:chat-id chat-id
+                                                         :chat    updated-chat})]}))
+(handlers/register-handler-fx
+ :signals/envelope-status
+ [re-frame/trim-v]
+ (fn [{:keys [db] :as cofx} [envelope-hash status]]
+   (let [{:keys [chat-id message-type message-id]}
+         (get-in db [:transport/message-envelopes envelope-hash])]
+     (case message-type
+       :contact-request
+       (if (= :sent status)
+         (handlers-macro/merge-fx cofx
+                                  (remove-hash envelope-hash)
+                                  (update-send-contact-request chat-id)))
+
+       (let [message (get-in db [:chats chat-id :messages message-id])
+             {:keys [fcm-token]} (get-in db [:contacts/contacts chat-id])]
+         (handlers-macro/merge-fx cofx
+                                  (remove-hash envelope-hash)
+                                  (models.message/update-message-status message status)
+                                  (models.message/send-push-notification fcm-token status)))))))
+
+(defn- own-info [db]
+  (let [{:keys [name photo-path address]} (:account/account db)
+        fcm-token (get-in db [:notifications :fcm-token])]
+    {:name          name
+     :profile-image photo-path
+     :address       address
+     :fcm-token     fcm-token}))
+
+(handlers/register-handler-fx
+ :contact/send-contact-request
+ (fn [{:keys [db] :as cofx} [_ chat-id]]
+   (when (and (= :online (:network-status db))
+              (pos? (count (:peers-summary db))))
+     (let [contact-request
+           (v1.contact/map->ContactRequest (own-info db))
+           {:keys [sym-key topic] :as chat}
+           (get-in db [:transport/chats chat-id])]
+       (message/send (v1.contact/NewContactKey. sym-key topic contact-request)
+                     chat-id cofx)))))
+
+(defn resend-contact-requests
+  ([cofx]
+   (resend-contact-requests [] cofx))
+  ([previous-summary {:keys [db]}]
+   (when (zero? (count previous-summary))
+     (let [contact-requests-fx (reduce (fn [acc [chat-id {:keys [send-contact-request?]}]]
+                                         (if send-contact-request?
+                                           (conj acc [:contact/send-contact-request chat-id])
+                                           acc))
+                                       []
+                                       (:transport/chats db))]
+       (when-not (empty? contact-requests-fx)
+         {:dispatch-n contact-requests-fx})))))
```

Comment:
No need for multiple dispatches there - `:shh` effect is already reworked so it can be accumulated just fine, therefore, `:contact/send-contact-request` handler should be extracted into standard cofx taking/fx producing function and we can just fold over `:transport/chats` with `merge-effects` helper function.